tests/bluetooth/tester/src/mesh.c

```diff
@@ -0,0 +1,305 @@
+		     net_idx, addr);
+
+	tester_send(BTP_SERVICE_ID_MESH, MESH_EV_PROVISIONED, CONTROLLER_INDEX,
+		    NULL, 0);
+}
+
+static void prov_reset(void)
+{
+	SYS_LOG_DBG("The local node has been reset and needs reprovisioning");
+}
+
+static const struct bt_mesh_comp comp = {
+	.cid = CID,
+	.elem = elements,
+	.elem_count = ARRAY_SIZE(elements),
+};
+
+static struct bt_mesh_prov prov = {
+	.uuid = dev_uuid,
+	.static_val = static_auth,
+	.static_val_len = sizeof(static_auth),
+	.output_number = output_number,
+	.output_string = output_string,
+	.input = input,
+	.link_open = link_open,
+	.link_close = link_close,
+	.complete = prov_complete,
+	.reset = prov_reset,
+};
+
+static void config_prov(u8_t *data, u16_t len)
+{
+	const struct mesh_config_provisioning_cmd *cmd = (void *) data;
+
+	memcpy(dev_uuid, cmd->uuid, sizeof(dev_uuid));
+	memcpy(static_auth, cmd->static_auth, sizeof(static_auth));
+
+	prov.output_size = cmd->out_size;
+	prov.output_actions = sys_le16_to_cpu(cmd->out_actions);
+	prov.input_size = cmd->in_size;
+	prov.input_actions = sys_le16_to_cpu(cmd->in_actions);
+
+	tester_rsp(BTP_SERVICE_ID_MESH, MESH_CONFIG_PROVISIONING,
+		   CONTROLLER_INDEX, BTP_STATUS_SUCCESS);
+}
+
+static void provision_node(u8_t *data, u16_t len)
+{
+	const struct mesh_provision_node_cmd *cmd = (void *) data;
+
+	memcpy(dev_key, cmd->dev_key, sizeof(dev_key));
+	memcpy(net_key, cmd->net_key, sizeof(net_key));
+
+	addr = sys_le16_to_cpu(cmd->addr);
+	flags = cmd->flags;
+	iv_index = sys_le32_to_cpu(cmd->iv_index);
+	net_key_idx = sys_le16_to_cpu(cmd->net_key_idx);
+	seq_num = sys_le32_to_cpu(cmd->seq_num);
+
+	tester_rsp(BTP_SERVICE_ID_MESH, MESH_PROVISION_NODE,
+		   CONTROLLER_INDEX, BTP_STATUS_SUCCESS);
+}
+
+static void init(u8_t *data, u16_t len)
+{
+	u8_t status = BTP_STATUS_SUCCESS;
+	int err;
+
+	err = bt_mesh_init(&prov, &comp);
+	if (err) {
+		status = BTP_STATUS_FAILED;
+
+		goto rsp;
+	}
+
+	if (addr) {
+		err = bt_mesh_provision(net_key, net_key_idx, flags, iv_index,
+					seq_num, addr, dev_key);
+		if (err) {
+			status = BTP_STATUS_FAILED;
+		}
+	} else {
+		err = bt_mesh_prov_enable(BT_MESH_PROV_ADV | BT_MESH_PROV_GATT);
+		if (err) {
+			status = BTP_STATUS_FAILED;
+		}
+	}
+
+rsp:
+	tester_rsp(BTP_SERVICE_ID_MESH, MESH_INIT, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+static void reset(u8_t *data, u16_t len)
+{
+	bt_mesh_reset();
+
+	tester_rsp(BTP_SERVICE_ID_MESH, MESH_RESET, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+static void input_number(u8_t *data, u16_t len)
+{
+	const struct mesh_input_number_cmd *cmd = (void *) data;
+	u8_t status = BTP_STATUS_SUCCESS;
+	int err;
+
+	err = bt_mesh_input_number(sys_le32_to_cpu(cmd->number));
+	if (err) {
+		status = BTP_STATUS_FAILED;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_MESH, MESH_INPUT_NUMBER, CONTROLLER_INDEX,
+		   status);
+}
+
+static void input_string(u8_t *data, u16_t len)
+{
+	const struct mesh_input_string_cmd *cmd = (void *) data;
+	char input_str[cmd->string_len + 1];
```

Comment:
I have a mixed feelings with those VLAs,  auth is limited to 16bytes anyway so just make this 16 + 1.  Aand for now just return error (or maybe simply truncate string, this is what bt_mesh_input_string would do anyway) if longer string was received.